addons/mail/models/mail_link_preview.py

```diff
@@ -38,19 +39,28 @@ def _create_from_message_and_notify(self, message):
             'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'
         })
         link_preview_values = []
+        link_preview_ids = []
+        link_previews_by_url = dict((preview.source_url, preview) for preview in message.link_preview_ids)
         for url in list(urls):
+            if url in link_previews_by_url:
+                preview = link_previews_by_url.pop(url)
+                if not preview.hide:
+                    link_preview_ids.append(preview.id)
+                continue
             if preview := link_preview.get_link_preview_from_url(url, requests_session):
                 preview['message_id'] = message.id
                 link_preview_values.append(preview)
-            if len(link_preview_values) > 5:
+            if len(link_preview_values) + len(link_preview_ids) > 5:
                 break
-        if link_preview_values:
-            link_previews = self.env['mail.link.preview'].create(link_preview_values)
+        for unused_preview in link_previews_by_url.values():
+            unused_preview._hide_and_notify()
+            unused_preview.unlink()
+        if link_preview_ids or link_preview_values:
             self.env['bus.bus']._sendone(message._bus_notification_target(), 'mail.record/insert', {
-                'LinkPreview': link_previews._link_preview_format()
+                'LinkPreview': self.browse(link_preview_ids + link_previews.create(link_preview_values).ids)._link_preview_format()
             })
 
-    def _unlink_and_notify(self):
+    def _hide_and_notify(self):
```

Comment:
Also I think `mail.link.preview/delete` could become a record/insert instead like the one for new.

Reply:
does `record/insert` handles `Record.delete()` ?

Reply:
ah, i see the `DELETE` command

Reply:
Yes for `DELETE` but it will work fully only if `onDelete: (r) => r.delete(),` is set as well.